packages/ckb-sdk-utils/src/crypto/bech32.ts

```diff
@@ -0,0 +1,148 @@
+const ALPHABET = 'qpzry9x8gf2tvdw0s3jn54khce6mua7l'
+
+const ALPHABET_MAP = new Map()
+
+for (let i = 0; i < ALPHABET.length; i++) {
+  const char = ALPHABET.charAt(i)
+  if (ALPHABET_MAP.get(char) !== undefined) {
+    throw new TypeError(`${char} is ambiguous`)
+  }
+  ALPHABET_MAP.set(char, i)
+}
+
+const polymodStep = (pre: any) => {
+  const b = pre >> 25
+  return (
+    ((pre & 0x1ffffff) << 5) ^
+    (-((b >> 0) & 1) & 0x3b6a57b2) ^
+    (-((b >> 1) & 1) & 0x26508e6d) ^
+    (-((b >> 2) & 1) & 0x1ea119fa) ^
+    (-((b >> 3) & 1) & 0x3d4233dd) ^
+    (-((b >> 4) & 1) & 0x2a1462b3)
+  )
+}
+
+const prefixChk = (prefix: string) => {
+  let chk = 1
+  for (let i = 0; i < prefix.length; ++i) {
+    const c = prefix.charCodeAt(i)
+    if (c < 33 || c > 126) throw new Error(`Invalid prefix (${prefix})`)
+
+    chk = polymodStep(chk) ^ (c >> 5)
+  }
+  chk = polymodStep(chk)
+
+  for (let i = 0; i < prefix.length; ++i) {
+    const v = prefix.charCodeAt(i)
+    chk = polymodStep(chk) ^ (v & 0x1f)
+  }
+  return chk
+}
+
+export const encode = (_prefix: string, words: Uint8Array, LIMIT: number = 90) => {
+  const prefix = _prefix.toLowerCase()
+
+  if (prefix.length + 7 + words.length > LIMIT) throw new TypeError('Exceeds length limit')
+
+  // determine chk mod
+  let chk = prefixChk(prefix)
+  let result = `${prefix}1`
+  for (let i = 0; i < words.length; ++i) {
+    const x = words[i]
+    if (x >> 5 !== 0) throw new Error('Non 5-bit word')
+
+    chk = polymodStep(chk) ^ x
+    result += ALPHABET.charAt(x)
+  }
+
+  for (let i = 0; i < 6; ++i) {
+    chk = polymodStep(chk)
+  }
+  chk ^= 1
+
+  for (let i = 0; i < 6; ++i) {
+    const v = (chk >> ((5 - i) * 5)) & 0x1f
+    result += ALPHABET.charAt(v)
+  }
+
+  return result
+}
+
+/* eslint-disable no-continue */
+export const decode = (_str: string, LIMIT: number = 90) => {
```

Comment:
`_str` => `str`, `LIMIT` => `limit`

Reply:
updated in https://github.com/nervosnetwork/ckb-sdk-js/pull/100/commits/54327f0c832e165a4f4b86fc0b5447ec9e54a81c